src/colvarcomp_distances.cpp

```diff
@@ -964,36 +1040,33 @@ int colvar::rmsd::init_permutation(std::string const &conf)
     }
   }
 
+  permutation_msds.assign(n_permutations, 0);
+
   return error_code;
 }
 
 
 void colvar::rmsd::calc_value()
 {
   // rotational-translational fit is handled by the atom group
-
-  x.real_value = 0.0;
-  for (size_t ia = 0; ia < atoms->size(); ia++) {
-    const cvm::atom_pos pos_ia(
-      atoms->pos_x(ia), atoms->pos_y(ia), atoms->pos_z(ia));
-    x.real_value += (pos_ia - ref_pos[ia]).norm2();
-  }
-  best_perm_index = 0;
-
-  // Compute sum of squares for each symmetry permutation of atoms, keep the smallest
-  size_t ref_pos_index = atoms->size();
-  for (size_t ip = 1; ip < n_permutations; ip++) {
-    cvm::real value = 0.0;
+  permutation_msds.assign(n_permutations, 0);
+  // x.real_value = 0.0;
+  for (size_t ip = 0; ip < n_permutations; ip++) {
+    const auto ref_pos_x = ref_pos_soa.begin() + ip * atoms->size();
+    const auto ref_pos_y = ref_pos_x + num_ref_pos;
+    const auto ref_pos_z = ref_pos_y + num_ref_pos;
```

Comment:
Nice refactor. I wonder if compilers unroll and vectorize this loop easily, or if we should make a pure SOA version of it, without loop, computing the distance between vectors of coordinates as stored in memory.